codegen/src/main/java/software/amazon/awssdk/codegen/poet/builder/BaseClientBuilderClass.java

```diff
@@ -623,4 +625,33 @@ private boolean hasClientContextParams() {
         Map<String, ClientContextParam> clientContextParams = model.getClientContextParams();
         return clientContextParams != null && !clientContextParams.isEmpty();
     }
+
+    private MethodSpec validateClientOptionsMethod() {
+        MethodSpec.Builder builder = MethodSpec.methodBuilder("validateClientOptions")
+                                               .addModifiers(PROTECTED, Modifier.STATIC)
+                                               .addParameter(SdkClientConfiguration.class, "c")
+                                               .returns(void.class);
+
+        if (AuthUtils.usesAwsAuth(model)) {
+            builder.addStatement("$T.notNull(c.option($T.SIGNER), $S)",
+                                 Validate.class,
+                                 SdkAdvancedClientOption.class,
+                                 "The 'overrideConfiguration.advancedOption[SIGNER]' must be configured in the client builder.");
+        }
+
+        if (AuthUtils.usesBearerAuth(model)) {
```

Comment:
Should it be `else if`? Can they use both aws auth and bearer auth?

Reply:
Yes they can. For example, the service `signatureVersion` has `v4` but also includes one operation that uses `bearer`.